include/boost/uuid/uuid_io.hpp

```diff
@@ -146,23 +146,32 @@ inline wchar_t to_wchar(size_t i) {
 
 } // namespace detail
 
-inline std::string to_string(uuid const& u)
+template<class OutputIterator>
+bool to_chars(uuid const& u, OutputIterator out, size_t buffer_size)
```

Comment:
`std::size_t`.

And it's not quite conventional to pass a size along with an output iterator. Doubly so since you are not outputting anything if `buffer_size < 36`.

I think this method should follow the convention of working with iterators, where you pass only the output iterator, which ensures there is enough storage. Another version specifically designed for raw buffers can be provided. That version would accept a raw pointer and a size of the buffer.

Reply:
Yes, it's a bit unconventional. Another overload is a good idea.